### PR TITLE
Grey out all context switches from processes other than the target

### PR DESCRIPTION
**Pull request by dpallotti**
Only color context switch slices related to the same process as the target.
Note that when selecting a (grey) context switch slice associated with another
process, all and only the other context switch slices associated with the same
thread are still colored.

A PID field has been added to ContextSwitch and Timer and this breaks
compatibility of the communication with previous builds.

Bug: b/152363779

### CODE REVIEW

File: OrbitGl/TimeGraph.cpp [hunk 1]
```diff
@@ -302,8 +302,8 @@ void TimeGraph::AddContextSwitch(const ContextSwitch& a_CS) {
           Timer timer;
           timer.m_Start = lastCS.m_Time;
           timer.m_End = a_CS.m_Time;
-          timer.m_TID = a_CS.m_ThreadId;
-          timer.m_Processor = (int8_t)a_CS.m_ProcessorIndex;
+          timer.m_TID = lastCS.m_ThreadId;
+          timer.m_Processor = (int8_t)lastCS.m_ProcessorIndex;
```
Comment (dimitry-) [inline OrbitGl/TimeGraph.cpp:308]: nit: static_cast<>

File: OrbitGl/TimeGraph.cpp [hunk 2]
```diff
@@ -302,8 +302,8 @@ void TimeGraph::AddContextSwitch(const ContextSwitch& a_CS) {
           Timer timer;
           timer.m_Start = lastCS.m_Time;
           timer.m_End = a_CS.m_Time;
-          timer.m_TID = a_CS.m_ThreadId;
-          timer.m_Processor = (int8_t)a_CS.m_ProcessorIndex;
+          timer.m_TID = lastCS.m_ThreadId;
```
Comment (dimitry-) [inline OrbitGl/TimeGraph.cpp:307]: maybe add an explanation from commit message here?

File: OrbitGl/TimeGraph.cpp [hunk 3]
```diff
@@ -323,7 +323,7 @@ void TimeGraph::AddContextSwitch(const ContextSwitch& a_CS) {
           Timer timer;
           timer.m_Start = lastCS.m_Time;
           timer.m_End = a_CS.m_Time;
-          timer.m_TID = a_CS.m_ThreadId;
+          timer.m_TID = lastCS.m_ThreadId;
```
Comment (dimitry-) [inline OrbitGl/TimeGraph.cpp:328]: and here

File: OrbitGl/TimeGraph.cpp [hunk 4]
```diff
@@ -583,7 +589,12 @@ void TimeGraph::UpdatePrimitives(bool a_Picking) {
 
           col = isSelected
```
Comment (dimitry-) [inline OrbitGl/TimeGraph.cpp:590]: this is a lot for :? maybe replace with if/else if else?

File: OrbitCore/ContextSwitch.cpp [hunk 5]
```diff
@@ -8,14 +8,15 @@
...
       m_Time(0),
       m_ProcessorIndex(0xFF),
       m_ProcessorNumber(0xFF) {}
 
 //-----------------------------------------------------------------------------
-ContextSwitch::~ContextSwitch() {}
+ContextSwitch::~ContextSwitch() = default;
 
 ORBIT_SERIALIZE(ContextSwitch, 0) {
   ORBIT_NVP_VAL(0, m_ThreadId);
```
Comment (dimitry-) [inline OrbitCore/ContextSwitch.cpp:22]: do you want to add m_ProcessId here?

File: OrbitGl/TimeGraph.cpp (hunk 1 above)
Comment (dpallotti) [reply OrbitGl/TimeGraph.cpp]: Done.

File: OrbitGl/TimeGraph.cpp (hunk 2 above)
Comment (dpallotti) [reply OrbitGl/TimeGraph.cpp:310]: Done.

File: OrbitGl/TimeGraph.cpp (hunk 3 above)
Comment (dpallotti) [reply OrbitGl/TimeGraph.cpp:335]: Done.

File: OrbitCore/ContextSwitch.cpp (hunk 5 above)
Comment (dpallotti) [reply OrbitCore/ContextSwitch.cpp:23]: Yep, thanks.

File: OrbitGl/TimeGraph.cpp (hunk 4 above)
Comment (dpallotti) [reply OrbitGl/TimeGraph.cpp]: Done.

**Pull request merged**